packages/blade/src/components/Button/BaseButton/__tests__/BaseButton.web.test.tsx

```diff
@@ -346,4 +349,47 @@ describe('<BaseButton />', () => {
     );
     expect(container).toMatchSnapshot();
   });
+
+  it('should have accessibilityLabel', () => {
+    const buttonText = 'Pay Now';
+    const { getByRole } = renderWithTheme(
+      <BaseButton accessibilityLabel="Close" icon={CloseIcon}>
+        {buttonText}
+      </BaseButton>,
+    );
+
+    expect(getByRole('button')).toHaveAccessibleName('Close');
+  });
+
+  it('should announce button loading accessibilityLabel', async () => {
```

Comment:
We can but it will just be a stub of the AccessibilityInfo, not sure if it's valuable to do so. 
We should trust that the platform will call it anyways.